### PR TITLE
Improvement of the MySQL Tables

### PR DESCRIPTION
**Pull request by BrainStone**
This PR aims to improve the MySQL storage, by using better suited data types and more indices

TODO List:

- [x] Improve table schema
- [x] Implement conversion of old tables
- [x] Test new table schema
- [x] Test table conversion

(The main reasoning behind opening this PR already is to allow you to check the progress and spot issues early on)

### CODE REVIEW

**Comment by supertassu** [comment]
Does this work with old (created before this) databases?

**Comment by BrainStone** [comment]
I'll add some code that'll convert old databases.

**Comment by BrainStone** [comment]
@supertassu So the migration has been added, but is untested. The querries themselves are tested on my end, but not through the plugin itself. I'll catch up on that testing in a little while.

File: src/main/java/me/leoko/advancedban/utils/SQLQuery.java [hunk 1]
```diff
@@ -7,17 +7,20 @@
...
-            "`operator` VARCHAR(16) NULL DEFAULT NULL," +
-            "`punishmentType` VARCHAR(16) NULL DEFAULT NULL," +
-            "`start` LONG DEFAULT NULL," +
-            "`end` LONG DEFAULT NULL," +
-            "`calculation` VARCHAR(50) NULL DEFAULT NULL," +
-            "PRIMARY KEY (`id`))",
+            "CREATE TABLE IF NOT EXISTS `Punishments` (" +
+            "`id` INT NOT NULL AUTO_INCREMENT," +
+            "`name` VARCHAR(16) NOT NULL," +
+            "`uuid` VARCHAR(35) NOT NULL," +
```
Comment (krusic22) [inline src/main/java/me/leoko/advancedban/utils/SQLQuery.java]: You should set this to VARCHAR(36) since that's the standard.

Comment (BrainStone) [reply src/main/java/me/leoko/advancedban/utils/SQLQuery.java]: Or 32. As the plugin stores the UUIDs without dashes anyways.

I’ll check the code to see what’s better.

**Comment by BrainStone** [comment]
I think at this point the PR is feature complete.

I'll get to testing it and will let you know once it's complete.

@krusic22 That should be better ;)

**Comment by BrainStone** [comment]
All tested and ready to merge!

**Pull request merged**